ConfigSpace/hyperparameters.pyx

```diff
@@ -877,12 +888,283 @@ cdef class NormalFloatHyperparameter(FloatHyperparameter):
+            else: 
+                return (ub - lb) / 2
+
+        elif self.is_legal(default_value):
+            return default_value
+        else:
+            raise ValueError("Illegal default value %s" % str(default_value))
+
+    def to_integer(self) -> 'BetaIntegerHyperparameter':
+        if self.q is None:
+            q_int = None
+        else:
+            q_int = int(np.rint(self.q))
+        
+        lower = int(np.ceil(self.lower))
+        upper = int(np.floor(self.upper))
+        default_value = int(np.rint(self.default_value))
+        return BetaIntegerHyperparameter(self.name, lower=lower, upper=upper, alpha=self.alpha, beta=self.beta,
+                                           default_value=int(np.rint(self.default_value)),
+                                           q=q_int, log=self.log)
+
+    def is_legal(self, value: Union[float]) -> bool:
+        if not (isinstance(value, float) or isinstance(value, int)):
+            return False
+        elif self.upper >= value >= self.lower:
+            return True
+        else:
+            return False
+
+    cpdef bint is_legal_vector(self, DTYPE_t value):
+        if self._upper >= value >= self._lower:
+            return True
+        else:
+            return False
+
+    def _sample(self, rs: np.random.RandomState, size: Optional[int] = None
+                ) -> Union[np.ndarray, float]:
+        alpha = self.alpha
+        beta = self.beta
+        lower = self._lower
+        upper = self._upper
+        return spbeta(alpha, beta, loc=lower, scale=upper-lower).rvs(size=size, random_state=rs)
+
+    cpdef np.ndarray _transform_vector(self, np.ndarray vector):
+        if np.isnan(vector).any():
+            raise ValueError('Vector %s contains NaN\'s' % vector)
+        if self.log:
+            vector = np.exp(vector)
+        if self.q is not None:
+            vector = np.rint(vector / self.q) * self.q
+        return vector
+
+    cpdef double _transform_scalar(self, double scalar):
+        if scalar != scalar:
+            raise ValueError('Number %s is NaN' % scalar)
+        if self.log:
+            scalar = math.exp(scalar)
+        if self.q is not None:
+            scalar = round(scalar / self.q) * self.q
+        return scalar
+        
+    def _inverse_transform(self, vector: Optional[np.ndarray]) -> Union[float, np.ndarray]:
+        if vector is None:
+            return np.NaN
+
+        if self.log:
+            vector = np.log(vector)
+        return vector
+
+    def get_neighbors(
+        self,
+        value: Any,
+        rs: np.random.RandomState,
+        number: int = 4,
+        transform: bool = False,
+        std: float = 0.2
+    ) -> List[float]:
+        neighbors = []  # type: List[float]
+        norm_std = std * (self._upper - self._lower)
+        while len(neighbors) < number:
+            neighbor = rs.normal(value, norm_std)  # type: float
```

Comment:
> Is "neighbour" related to some distance dependant on the distribution or are we okay with just normally distributed numbers centered on value? I.e. How "neighbouring" something is defined by a simple absolute distance.

We sample neighbors in the unit hypercube. Since they are transformed into the real distribution afterwards, this transformation takes care of introducing this distribution-related distance.

Having said that, this function doesn't sample from the unit hypercube (which, frankly, isn't documented that we should do so). @hvarfner could you please update this function to treat value as the value in the unit hypercube and sample in that?

I'm wondering whether you should consider subclassing `UniformFloatHyperparameter`, then you could you the parent class implementation.